src/wp-includes/class-walker-nav-menu.php

```diff
@@ -276,4 +307,24 @@ public function end_el( &$output, $data_object, $depth = 0, $args = null ) {
 		$output .= "</li>{$n}";
 	}
 
+	/**
+	 * Builds an HTML attributes string based on an array of key value pairs
+	 * Empty values are ignored
+	 *
+	 * @since 6.1.2
+	 *
+	 * @param      array   $atts   The atts
+	 * @return     string  HTML attributes string
+	 */
+	protected function build_atts( $atts = array() ) {
+		$attributes = '';
+		foreach ( $atts as $attr => $value ) {
+			if ( is_scalar( $value ) && '' !== $value && false !== $value ) {
+				$value       = ( 'href' === $attr ) ? esc_url( $value ) : esc_attr( $value );
+				$attributes .= ' ' . $attr . '="' . $value . '"';
```

Comment:
```suggestion
				$attributes .= ' ' . esc_attr( $attr ) . '="' . $value . '"';
```
Attribute names will also need to be escaped here.

Reply:
Hi. I don't think this is applicable here. Attributes values and names have different possible characters so the escaping mechanism should be different. WordPress doesn't have an attribute name escaping mecanism yet so we should avoid adding it here since [it wasn't there before](https://github.com/WordPress/wordpress-develop/blob/d2f420493d87465c323190892d4e7a6e4c6070a7/src/wp-includes/class-walker-nav-menu.php#L209-L215). It could very well introduce breaking change to millions of users. I'll add some links down below to document previous conversations that happened about this in the last 10 years on WP core.